addons/udes_stock/models/stock_quant.py

```diff
@@ -136,3 +137,59 @@ def get_info(self):
     def quant_location_policy(self):
         self.ensure_one()
         self.location_id.apply_quant_policy()
+
+    def get_movelines_from_quant(self, aux_domain=None):
+        """Get the movelines associated to a quant
+        :param aux_domain: Extra domain arguments to add to the search
+        :returns: Associated movelines
+        """
+        self.ensure_one()
+        MoveLine = self.env["stock.move.line"]
+        domain = [
+            ("product_id", "=", self.product_id.id),
+            ("package_id", "=", self.package_id.id),
+            ("location_id", "=", self.location_id.id),
+            ("lot_id", "=", self.lot_id.id),
+            ("owner_id", "=", self.owner_id.id),
+        ]
+        if aux_domain is not None:
+            domain.extend(aux_domain)
+        return MoveLine.search(domain)
+
+    @api.multi
+    def _split(self, qty):
+        """Split a quant into two quants, a new quant with quantity `qty` and the original quantity of
+        self reduced by qty. If qty > quantity, it just returns self (original quant).
+        It gives the new quantity priority over reserved quantities
+        :param qty: Quantity to split from self, cretaing a new quant of quantity `qty`
+        :return: new quant or self if it cannot be split
+        """
+        self.ensure_one()
+        rounding = self.product_id.uom_id.rounding
+        if (
+            float_compare(abs(self.quantity), abs(qty), precision_rounding=rounding) <= 0
+        ):  # if quant <= qty in abs, take it entirely
+            return self
+        # Update the quantities
+        new_qty_round = float_round(qty, precision_rounding=rounding)
+        qty_round = float_round(self.quantity - qty, precision_rounding=rounding)
+        # Handle reserved quants
+        reserved_quantity = self.reserved_quantity
+        new_qty_round_reserved = float_round(
+            self.reserved_quantity - qty, precision_rounding=rounding
+        )
+        new_quant = self.sudo().copy(
+            default={
+                "quantity": new_qty_round,
+                "reserved_quantity": new_qty_round
+                if new_qty_round_reserved > 0
+                else reserved_quantity,
+            }
+        )
+        self.sudo().write(
+            {
+                "quantity": qty_round,
+                "reserved_quantity": qty_round if new_qty_round_reserved > 0 else 0,
+            }
+        )
```

Comment:
The handling of `reserved_quantity` does not look right.  

If 1 item is split from a quant of 5, with 3 reserved, we end up with:
```
{ qty: 4, reserved: 4 }
{ qty: 1, reserved: 1 }
```